### PR TITLE
make all simulation airframes simulator specific

### PR DESCRIPTION
**Pull request by dagar**
Resurrecting https://github.com/PX4/PX4-Autopilot/pull/20139.

 - all posix simulation airframes are now specific to a simulator and model and now named `${SIMULATOR}_${MODEL}`
 - added cmake warnings for all SITL target helpers that the relevant airframe exists (this caught a few leftovers)
 - jmavsim launched directly from px4 binary now (you can debug directly, etc)

``` Console
ROMFS/px4fmu_common/init.d-posix/airframes/
├── 10016_gazebo_iris
├── 10017_jmavsim_iris
├── 10018_gazebo_iris_foggy_lidar
├── 10019_gazebo_omnicopter
├── 10030_gazebo_px4vision
├── 10040_sihsim_quadx
├── 10041_sihsim_airplane
├── 10042_sihsim_xvert
├── 1010_gazebo_iris_opt_flow
├── 1010_gazebo_iris_opt_flow.post
├── 1011_gazebo_iris_irlock
├── 1012_gazebo_iris_rplidar
├── 1013_gazebo_iris_vision
├── 1013_gazebo_iris_vision.post
├── 1015_gazebo_iris_obs_avoid
├── 1015_gazebo_iris_obs_avoid.post
├── 1017_gazebo_iris_opt_flow_mockup
├── 1019_gazebo_iris_dual_gps
├── 1021_gazebo_uuv_hippocampus
├── 1022_gazebo_uuv_bluerov2_heavy
├── 1030_gazebo_plane
├── 1031_gazebo_plane_cam
├── 1032_gazebo_plane_catapult
├── 1033_jsbsim_rascal
├── 1034_flightgear_rascal-electric
├── 1035_gazebo_techpod
├── 1036_jsbsim_malolo
├── 1037_gazebo_believer
├── 1038_gazebo_glider
├── 1039_flightgear_rascal
├── 1039_gazebo_advanced_plane
├── 1040_gazebo_standard_vtol
├── 1041_gazebo_tailsitter
├── 1042_gazebo_tiltrotor
├── 1043_gazebo_standard_vtol_drop
├── 1044_gazebo_plane_lidar
├── 1060_gazebo_rover
├── 1061_gazebo_r1_rover
├── 1062_flightgear_tf-r1
├── 1070_gazebo_boat
├── 17001_flightgear_tf-g1
├── 17002_flightgear_tf-g2
├── 2507_gazebo_cloudship
├── 3010_jsbsim_quadrotor_x
├── 3011_jsbsim_hexarotor_x
├── 4001_gz_x500
├── 4002_gz_x500_depth
├── 6011_gazebo_typhoon_h480
├── 6011_gazebo_typhoon_h480.post
└── CMakeLists.txt

0 directories, 50 files
```

### CODE REVIEW

**Review by Jaeyoung-Lim — APPROVED** [review]
Thanks!

There is a CI test failure, which I am not sure if it is related to this PR

**Comment by dagar** [comment]
> There is a CI test failure, which I am not sure if it is related to this PR

It was real, the MAVSDK tests needed to launch with `PX4_SIM_MODEL=gazebo_iris` (not just "iris").

**Comment by bperseghetti** [comment]
I still think it can't just say gazebo, it needs to say classic somewhere. Like classic-gazebo, gazebo classic, cgz, etc.

File: src/modules/simulation/gz_bridge/CMakeLists.txt [hunk 1]
```diff
@@ -82,21 +82,74 @@ if(gz-transport_FOUND)
...
-					COMMAND ${CMAKE_COMMAND} -E env PX4_SIM_MODEL=${model} $<TARGET_FILE:px4>
+					COMMAND ${CMAKE_COMMAND} -E env PX4_SIM_MODEL=gz_${model} $<TARGET_FILE:px4>
 					WORKING_DIRECTORY ${SITL_WORKING_DIR}
 					USES_TERMINAL
 					DEPENDS px4
 				)
 			else()
 				add_custom_target(gz_${model}_${world_name}
-					COMMAND ${CMAKE_COMMAND} -E env PX4_GZ_MODEL=${model} PX4_GZ_WORLD=${world_name} $<TARGET_FILE:px4>
+					COMMAND ${CMAKE_COMMAND} -E env PX4_SIM_MODEL=gz_${model} PX4_SIM_WORLD=${world_name} $<TARGET_FILE:px4>
```
Comment (beniaminopozzan) [inline src/modules/simulation/gz_bridge/CMakeLists.txt:152]: @dagar , to define the world file `PX4_GZ_WORLD` is need, not `PX4_SIM_WORLD`.

Comment (dagar) [reply src/modules/simulation/gz_bridge/CMakeLists.txt:152]: Hmm, this is getting kind of convoluted, maybe we need to back up a bit and review/document it centrally before it gets any worse.

Comment (beniaminopozzan) [reply src/modules/simulation/gz_bridge/CMakeLists.txt:152]: Well, at least for gazebo ignition is it quite documented. Do you mean adding docs here or in https://docs.px4.io/?

And what is the role of `PX4_SIM_WORLD`? It is defined here for classic gazebo https://github.com/PX4/PX4-Autopilot/blob/main/Tools/simulation/gazebo/sitl_run.sh#L71 and here for jsbsim https://github.com/PX4/PX4-Autopilot/blob/main/Tools/simulation/jsbsim/sitl_run.sh#L41 but it does not appear to be used.

Comment (dagar) [reply src/modules/simulation/gz_bridge/CMakeLists.txt:152]: > And what is the role of PX4_SIM_WORLD

Yes that's what I'm getting at.

**Comment by Jaeyoung-Lim** [comment]
@dagar Does this mean that `make px4_sitl gazebo` is no longer a valid build target?

**Comment by bperseghetti** [comment]
Would sorta hope that it would be gazebo_classic instead of sitl_gazebo. In my mind sitl is a redundant prefix for anything POSIX simulation related especially for classic gazebo or gazebo (gz).

**Comment by Jaeyoung-Lim** [comment]
@bperseghetti I agree that `gazebo` is confusing, but `gazebo_classic` would be a breaking change, and would require an overhaul of the documentation without much gain. 

would you say still the distinction between `gazebo` and `gz` is not enough?

**Comment by bperseghetti** [comment]
@Jaeyoung-Lim Perfect, a "breaking change" is what is exactly needed. We need to let people know classic gazebo is dying/dead, and they need to move on.  I think we could maybe catch people using "gazebo" and redirect them to a standard " did you mean "gz" or "gazebo_classic" prompt. Maybe even include a link explaining it?

**Comment by bperseghetti** [comment]
@Jaeyoung-Lim considering that OSRF has decided to name what we call gz as gazebo and what we call gazebo as gazebo_classic...... I think that we need to follow the decisions made.... For better or for worse sitl_gazebo is not gazebo anymore... It's gazebo_classic. It will be dead in a year. And it was never meant to even be installable on 22.04, that's a packaging fluke. I would suggest, let's start the process of creating less confusion, instead of sowing needless wasted effort for what? Someone who can't read a terminal output? I already have a branch that has begun all the docs renaming, instead of creating another meaningless PR, I personally believe we should actually follow the naming decision from the tools we use. Because no matter what at some point we have to anyhow. I also might be 5 screwdrivers deep while on a red-eye flight back from CES... But, I think we need to rip the band-aid off now or risk even more confusion.

**Pull request merged**